### PR TITLE
Implementing basic daemon with configuration

### PR DESCRIPTION
**Pull request by taktv6**

### CODE REVIEW

File: cmd/bio-rd/config/bgp.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+	"fmt"
+	"time"
+
+	bnet "github.com/bio-routing/bio-rd/net"
+	"github.com/bio-routing/bio-rd/routingtable/filter"
+	"github.com/pkg/errors"
+)
+
+type BGP struct {
+	Groups []*BGPGroup `yaml:"groups"`
```
Comment (hikhvar) [inline cmd/bio-rd/config/bgp.go]: Why a slice of pointers?

File: cmd/bio-rd/config/policy.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	if pst.Then.NextHop != nil {
+		addr, err := bnet.IPFromString(pst.Then.NextHop.Address)
+		if err != nil {
+			return nil, errors.Wrap(err, "Invalid next_hop address")
+		}
+
+		a = append(a, actions.NewSetNextHopAction(addr))
+	}
+
+	if pst.Then.Accept {
```
Comment (hikhvar) [inline cmd/bio-rd/config/policy.go]: What about adding an error if both Accept and Reject are set in the same term? IMHO this is most likely a configuration error.

File: cmd/bio-rd/main.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+				bgpSrv.ReplaceImportFilterChain(n.PeerAddressIP, newCfg.IPv4.ImportFilterChain)
+				bgpSrv.ReplaceExportFilterChain(n.PeerAddressIP, newCfg.IPv4.ExportFilterChain)
+				continue
+			}
+
+			bgpSrv.DisposePeer(oldCfg.PeerAddress)
+		}
+	}
+
+	// Turn up all sessions that are missing
```
Comment (hikhvar) [inline cmd/bio-rd/main.go]: Thats all missing sessions and the sessions teared down above due to NeedsRestart?

File: protocols/bgp/packet/path_attributes.go [hunk 4]
```diff
@@ -817,6 +817,17 @@ func PathAttributes(p *route.Path, iBGP bool, rrClient bool) (*PathAttribute, er
 	last.Next = nextHop
 	last = nextHop
 
+	if p.BGPPath.MED != 0 {
```
Comment (hikhvar) [inline protocols/bgp/packet/path_attributes.go]: Do we have a testcase regarding the proper rendering with MED?

File: routingtable/filter/term_condition.go [hunk 5]
```diff
@@ -101,3 +101,29 @@ func (t *TermCondition) matchesLargeCommunityFilters(pa *route.Path) bool {
...
+		return false
+	}
+
+	for i := range t.routeFilters {
+		if !t.routeFilters[i].equal(x.routeFilters[i]) {
+			return false
+		}
+	}
+
+	// TODO: Compare community filters
```
Comment (hikhvar) [inline routingtable/filter/term_condition.go]: Lets create an issue after merge, to track this TODO

File: protocols/bgp/packet/path_attributes.go (hunk 4 above)
Comment (taktv6) [reply protocols/bgp/packet/path_attributes.go]: No.

File: routingtable/filter/term_condition.go (hunk 5 above)
Comment (taktv6) [reply routingtable/filter/term_condition.go]: Ack.

File: cmd/bio-rd/main.go (hunk 3 above)
Comment (taktv6) [reply cmd/bio-rd/main.go]: Correct.

File: cmd/bio-rd/config/bgp.go (hunk 1 above)
Comment (taktv6) [reply cmd/bio-rd/config/bgp.go]: BGPGroup has pointer receivers

File: cmd/bio-rd/config/policy.go (hunk 2 above)
Comment (taktv6) [reply cmd/bio-rd/config/policy.go]: Done.

File: routingtable/filter/term_condition.go (hunk 5 above)
Comment (taktv6) [reply routingtable/filter/term_condition.go]: https://github.com/bio-routing/bio-rd/issues/216

**Pull request merged**